wp-includes/service-workers.php

```diff
@@ -37,6 +37,64 @@ function wp_register_service_worker( $handle, $src, $deps = array(), $scope = WP
 	return wp_service_workers()->register( $handle, $src, $deps, $scope );
 }
 
+/**
+ * Register route and caching strategy using regex pattern for route.
+ *
+ * @since 0.2
+ *
+ * @param string $route Route, has to be valid regex.
+ * @param string $strategy Strategy, can be WP_Service_Workers::STRATEGY_STALE_WHILE_REVALIDATE, WP_Service_Workers::STRATEGY_CACHE_FIRST,
+ *                         WP_Service_Workers::STRATEGY_NETWORK_FIRST, WP_Service_Workers::STRATEGY_CACHE_ONLY,
+ *                         WP_Service_Workers::STRATEGY_NETWORK_ONLY.
+ * @param array  $strategy_args {
+ *     An array of strategy arguments.
+ *
+ *     @type string $cache_name Cache name.
+ *     @type array  $plugins    Array of plugins with configuration. The key of each plugin in the array must match the plugin's name.
+ *                              See https://developers.google.com/web/tools/workbox/guides/using-plugins#workbox_plugins.
+ * }
+ */
+function wp_register_caching_strategy_by_route_pattern( $route, $strategy = WP_Service_Workers::STRATEGY_STALE_WHILE_REVALIDATE, $strategy_args = array() ) {
+	wp_service_workers()->register_cached_route_pattern( $route, $strategy, $strategy_args );
+}
+
+/**
+ * Register route and caching strategy for URL.
+ *
+ * @since 0.2
+ *
+ * @param string $route Route, has to be string literal.
+ * @param string $strategy Strategy, can be WP_Service_Workers::STRATEGY_STALE_WHILE_REVALIDATE, WP_Service_Workers::STRATEGY_CACHE_FIRST,
+ *                         WP_Service_Workers::STRATEGY_NETWORK_FIRST, WP_Service_Workers::STRATEGY_CACHE_ONLY,
+ *                         WP_Service_Workers::STRATEGY_NETWORK_ONLY.
+ * @param array  $strategy_args {
+ *     An array of strategy arguments.
+ *
+ *     @type string $cache_name Cache name.
+ *     @type array  $plugins    Array of plugins with configuration. The key of each plugin in the array must match the plugin's name.
+ *                              See https://developers.google.com/web/tools/workbox/guides/using-plugins#workbox_plugins.
+ * }
+ */
+function wp_register_caching_strategy_by_route_url( $route, $strategy = WP_Service_Workers::STRATEGY_STALE_WHILE_REVALIDATE, $strategy_args = array() ) {
+	wp_service_workers()->register_cached_route_url( $route, $strategy, $strategy_args );
```

Comment:
I just recalled that in WordPress the rewrite API and the REST API both only support regular expressions. That being the case, we could just opt to not expose registering non-regex routes in Workbox. As [the docs say](https://developers.google.com/web/tools/workbox/guides/route-requests#matching_a_route_with_a_string):

> Matching a route with a string is the easiest to understand, but also the least flexible option.

So if we just support regular expressions, then this would simplify the PHP API while also being more aligned with functions like `register_rest_route()` and `add_rewrite_rule()`.